src/mbgl/style/expression/value.cpp

```diff
@@ -8,37 +8,40 @@ namespace style {
-                       [&](const Image&) -> type::Type { return type::Image; },
-                       [&](const NullValue&) -> type::Type { return type::Null; },
-                       [&](const std::unordered_map<std::string, Value>&) -> type::Type { return type::Object; },
-                       [&](const std::vector<Value>& arr) -> type::Type {
-                           std::optional<type::Type> itemType;
-                           for (const auto& item : arr) {
-                               const type::Type t = typeOf(item);
-                               if (!itemType) {
-                                   itemType = {t};
-                               } else if (*itemType == t) {
-                                   continue;
-                               } else {
-                                   itemType = {type::Value};
-                                   break;
-                               }
-                           }
-
-                           return type::Array(itemType.value_or(type::Value), arr.size());
-                       });
+    return value.match(
+        [&](bool) -> type::Type { return type::Boolean; },
+        [&](double) -> type::Type { return type::Number; },
+        [&](const std::string&) -> type::Type { return type::String; },
+        [&](const Color&) -> type::Type { return type::Color; },
+        [&](const Padding&) -> type::Type { return type::Padding; },
+        [&](const VariableAnchorOffsetCollection&) -> type::Type { return type::VariableAnchorOffsetCollection; },
+        [&](const Collator&) -> type::Type { return type::Collator; },
+        [&](const Formatted&) -> type::Type { return type::Formatted; },
+        [&](const Image&) -> type::Type { return type::Image; },
+        [&](const NullValue&) -> type::Type { return type::Null; },
+        [&](const std::unordered_map<std::string, Value>&) -> type::Type { return type::Object; },
+        [&](const std::vector<Value>& arr) -> type::Type {
+            std::optional<type::Type> itemType;
+            for (const auto& item : arr) {
+                const type::Type t = typeOf(item);
+                if (!itemType) {
+                    itemType = {t};
+                } else if (*itemType == t) {
+                    continue;
+                } else {
+                    itemType = {type::Value};
+                    break;
+                }
+            }
+
+            return type::Array(itemType.value_or(type::Value), arr.size());
+        });
 }
 
 std::string toString(const Value& value) {
     return value.match([](const NullValue&) { return std::string(); },
-                       [](const Color& c) { return c.stringify(); }, // avoid quoting
+                       [](const Color& c) { return c.stringify(); },                         // avoid quoting
+                       [](const VariableAnchorOffsetCollection& v) { return v.toString(); }, // avoid quoting
```

Comment:
nit: duplicated comment?

Reply:
no, will revert the change in Color line next commit.

Reply:
It was change when do clang format